mux/bindings/conformance/runner.py

```
@@ -0,0 +1,325 @@
+    if isinstance(expected, dict):
+        if not isinstance(actual, dict):
+            return False
+        return all(key in actual and partial_match(actual[key], value) for key, value in expected.items())
+    if isinstance(expected, list):
+        if not isinstance(actual, list) or len(actual) < len(expected):
+            return False
+        return all(partial_match(actual[index], value) for index, value in enumerate(expected))
+    return actual == expected
+
+
+def expect_events(stream: Any, expected: List[Dict[str, Any]], timeout: float) -> None:
```

Ruff (0.15.20)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `stream`

(ANN401)